cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories_auth.go

```diff
@@ -0,0 +1,244 @@
+// Copyright 2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	apprepov1alpha1 "github.com/vmware-tanzu/kubeapps/cmd/apprepository-controller/pkg/apis/apprepository/v1alpha1"
+	corev1 "github.com/vmware-tanzu/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1"
+	"github.com/vmware-tanzu/kubeapps/cmd/kubeapps-apis/plugins/pkg/statuserror"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	k8scorev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	SecretCaKey         = "ca.crt"
+	SecretAuthHeaderKey = "authorizationHeader"
+)
+
+func secretNameForRepo(repoName string) string {
+	return fmt.Sprintf("apprepo-%s", repoName)
+}
+
+func newLocalOpaqueSecret(ownerRepo types.NamespacedName) *k8scorev1.Secret {
+	return &k8scorev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretNameForRepo(ownerRepo.Name),
+		},
+		Type: k8scorev1.SecretTypeOpaque,
+		Data: map[string][]byte{},
+	}
+}
+
+// this func is only used with kubeapps-managed secrets
+func newSecretFromTlsConfigAndAuth(repoName types.NamespacedName,
```

Comment:
Didn't we have a shared `/pkg/` folder share code across plugins? 
I'm not sure, but I'd rather remove any dup code if possible, no?
_(mentioning this bc I saw a similar fn in the flux plg)_

Reply:
Some code here is copied from Greg's Flux.
Most of the times due to Flux-specific structs it isn't possible to make it generic. I tried at the beginning but had to undo down the road.
We could do some refactoring in a later iteration if needed.